src/cljs/athens/views/blocks.cljs

```diff
@@ -221,7 +224,16 @@
 (defn on-change
   [value uid]
   ;; (prn "ONCHANGE" value)
-  (dispatch [:transact [{:db/id [:block/uid uid] :block/string value :edit/time (now-ts)}]]))
+  (dispatch [:transact [{:db/id [:block/uid uid] :block/string value :edit/time (now-ts)}]])
+  ;; automatically add non-existent pages
+  ;; TODO: delete pages that are no longer connected to anything else
+  (parse/transform {:page-link (fn [& title]
+                                 (if-not (db/search-exact-node-title (apply + title)) (let [uid (gen-block-uid)]
```

Comment:
This is a formatting thing: let the `then` and `else` clauses be separate lines. It also be clearer to flip the clauses and use normal `if`.

Reply:
I'm not sure how I feel about the page creation happening here, but fine to have it here for now.

Reply:
Can you explain the usage of `+ apply` to me? If it is because `title` comes in as a list with ` & title`, I would first like greater clarity or comments on why we use `&` in the first place.

Reply:
I agree that we should have some further discussion about where to process the new page creation/deletion logic. But I think it's more of a performance issue so we could just have the functionality now and talk about it later :)

Reply:
regarding the `+ apply` issue, because the nested page-link is processed recursively, `parse/transform` performs a leaf-first search which creates the page and returns the "unparsed" value of the `:page-link` node. For example, parsing `[[A[[B]]]]` would create a page named `B` but also returns `[[B]]` for its parent node to create (`A[[B]]`). `+ apply` is used to combine the list of children strings to the actual page name (a unique transformation).

Reply:
Wow, pretty technical issue explained pretty well!

Reply:
Also please let me resolve the comments going forward, I had to find and open this to find your response :)